### PR TITLE
Explore: Fix parsing absolute range when URL changes

### LINKED ISSUE #72497: Explore: Time range is incorrect when going back in browser history
### Steps to reproduce

1. Open Explore 
2. Run a query returning time series
3. Zoom in on the graph, time range will change to absolute
4. Zoom in once again, time range will be updated
5. Go back in browser history

Expected: Explore shows graph from step (3)
Actual: No graph is shown and the date range is set to invalid

### Technical details

When going back in the history, the URL `from` / `to` values are always passed as strings. RawTimeRange supports using DateTime or `string`, but `string` should be used only for relative, not absolute range. When the value in the URL is an absolute time in ms, it should be converted to DateTime object to conform to RawTimeRange type (string can be used only for relative time OR if the date is ISO-formatted, pure ms as strings are not supported).

The bug does not happen when Explore is initialized because the URL is parsed in initializeExplore action ([code](https://github.com/grafana/grafana/blob/v10.1.x/public/app/features/explore/state/explorePane.ts#L147)). However, when we update range from URL the parser simply passes through string value ([code](https://github.com/grafana/grafana/blob/v10.1.x/public/app/features/explore/hooks/useStateSync/migrators/v1.ts#L87))

### Additional info

**Environment**:
- Grafana version: 10.1.0
- User OS & Browser: any

### PR DESCRIPTION
**Pull request by ifrost**
Fixes: #72497

The main problem is that the range in the URL is to `ms` as a string. This, however, even though it matches the type of `RawTimeRange`, it's not compatible with that type. In `RawTimeRange` properties `from` and `to` can be `DateTime` or `string`, but `string` should be compatible with moment.js, meaning it has be to formatted with ISO, cannot be a string in "ms" (check example t https://jsfiddle.net/nobu222/uLrfhncw/).

The bug happens only when time range is updated in the URL (when history changes) because initialize Explore handles this indirectly [here](https://github.com/grafana/grafana/blob/v10.1.x/public/app/features/explore/state/explorePane.ts#L147C16-L147C24) (getRange does additional parsing, but we need to for updateTime from URL as well). 

Ideally we may want to define ExploreURL type a bit better, and improve the readability so it's clearer when and why the parsing happens but that proves to be a refactoring going beyond that simple fix, so I'll open a separate PR later (https://github.com/grafana/grafana/pull/72498 for more details)

### CODE REVIEW

File: public/app/features/explore/hooks/useStateSync/index.test.tsx [hunk 1]
```diff
@@ -396,6 +396,40 @@ describe('useStateSync', () => {
     expect(store.getState().explore.panes['one']?.range.raw).toMatchObject({ from: 'now-6h', to: 'now' });
   });
 
+  it('Changes time rage when the range in the URL is updated to absolute range', async () => {
```
Comment (harisrozajac) [inline public/app/features/explore/hooks/useStateSync/index.test.tsx:399]:
```suggestion
  it('Changes time range when the range in the URL is updated to absolute range', async () => {
```

**Review by harisrozajac — APPROVED** [review]
This looks great! Great fix! 🚀

File: public/app/features/explore/state/utils.ts [hunk 2]
```diff
@@ -157,6 +157,42 @@ export function getRange(range: RawTimeRange, timeZone: TimeZone): TimeRange {
...
+    to: parseRawTime(range.to),
+  };
+  if (parsedRange.from !== null && parsedRange.to !== null) {
+    rawTimeRange = { from: parsedRange.from, to: parsedRange.to };
+  }
+  return rawTimeRange;
+}
+
+/**
+ * Docs
```
Comment (Elfo404) [inline public/app/features/explore/state/utils.ts:176]: leftover?
```suggestion
```

**Review by Elfo404 — COMMENTED** [review]
can we also update `docs/sources/explore/_index.md`?

the schema is mistakenly defined as

```
{
  datasource: string; // the pane's root datasource UID, or `-- Mixed --` for mixed datasources
  queries: {
    refId: string; // an alphanumeric identifier for this query, must be unique within the pane, i.e. "A", "B", "C", etc.
    datasource: {
      uid: string; // the query's datasource UID ie: "AD7864H6422"
      type: string; // the query's datasource type-id, i.e: "loki"
    }
    // ... any other datasource-specific query parameters
  }[]; // array of queries for this pane
  range: {
    from: string | number; // the start time, in milliseconds since epoch
    to: string | number; // the end time, in milliseconds since epoch
  }
}
```

while in fact it should be 
```
{
  datasource: string; // the pane's root datasource UID, or `-- Mixed --` for mixed datasources
  queries: {
    refId: string; // an alphanumeric identifier for this query, must be unique within the pane, i.e. "A", "B", "C", etc.
    datasource: {
      uid: string; // the query's datasource UID ie: "AD7864H6422"
      type: string; // the query's datasource type-id, i.e: "loki"
    }
    // ... any other datasource-specific query parameters
  }[]; // array of queries for this pane
  range: {
    from: string; // the start time, in milliseconds since epoch
    to: string; // the end time, in milliseconds since epoch
  }
}
```

**Review by Elfo404 — APPROVED** [review]

**Pull request merged**